core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -197,11 +226,111 @@ contract PricelessPositionManager is Testable {
+
+            // Reset the position state as all the value has been removed after settlement.
+            delete positions[msg.sender];
+        }
+
+        // Transfer tokens and collateral.
+        require(
+            collateralCurrency.transfer(msg.sender, totalRedeemableCollateral.rawValue),
+            "Collateral redemption send failed"
+        );
+        require(
+            tokenCurrency.transferFrom(msg.sender, address(this), tokensToRedeem.rawValue),
+            "Token redemption send failed"
+        );
+        tokenCurrency.burn(tokensToRedeem.rawValue);
+
+        // Decrement total contract collateral and oustanding debt.
+        totalPositionCollateral = totalPositionCollateral.sub(totalRedeemableCollateral);
+        totalTokensOutstanding = totalTokensOutstanding.sub(tokensToRedeem);
+    }
+
+    function _deleteSponsorPosition(address sponsor) internal {
+        PositionData storage positionToLiquidate = _getPositionData(sponsor);
+
+        // Remove the collateral and outstanding from the overall total position.
+        totalPositionCollateral = totalPositionCollateral.sub(positionToLiquidate.collateral);
+        totalTokensOutstanding = totalTokensOutstanding.sub(positionToLiquidate.tokensOutstanding);
+
+        // Reset the sponsors position to have zero outstanding and collateral.
+        delete positions[sponsor];
+    }
+
     function _getPositionData(address sponsor) internal view returns (PositionData storage position) {
         position = positions[sponsor];
         require(position.isValid, "Position does not exist");
     }
 
+    function _requestOraclePrice(uint requestedTime) internal {
+        OracleInterface oracle = OracleInterface(_getOracleAddress());
+        oracle.requestPrice(priceIdentifer, requestedTime);
+    }
+
+    function _getOraclePrice(uint requestedTime) public view returns (FixedPoint.Unsigned memory) {
+        // Create an instance of the oracle and get the price. If the price is not resolved revert.
+        OracleInterface oracle = OracleInterface(_getOracleAddress());
+        require(oracle.hasPrice(priceIdentifer, requestedTime), "Can only get a price once the DVM has resolved");
+        int oraclePrice = oracle.getPrice(priceIdentifer, requestedTime);
+
+        // For now we don't want to deal with negative prices in positions.
+        if (oraclePrice < 0) {
+            oraclePrice = 0;
+        }
+        return FixedPoint.Unsigned(_safeUintCast(oraclePrice));
+    }
+
+    function _getOracleAddress() internal view returns (address) {
+        bytes32 oracleInterface = "Oracle";
+        return finder.getImplementationAddress(oracleInterface);
+    }
+
+    function _getIdentifierWhitelistAddress() internal view returns (address) {
+        bytes32 identifierWhitelistInterface = "IdentifierWhitelist";
+        return finder.getImplementationAddress(identifierWhitelistInterface);
+    }
+    function _getStoreAddress() internal view returns (address) {
```

Comment:
nit: is this used anywhere?

Reply:
not yet...Can remove it :)

Reply:
I would remove because we don't know if either of these are going to be used in this class. For instance, I think the store may only be used in the FeePayer class (depending on how we implement it).

Reply:
So it may have the effect of just increasing code churn.

Reply:
cool. will ditch